src/libs/actions/Report.js

```diff
@@ -983,6 +983,48 @@ function addAction(reportID, text, file) {
         .then(({reportAction}) => updateReportWithNewAction(reportID, reportAction));
 }
 
+/**
+ * Deletes a comment from the report, basically sets it as empty string
+ *
+ * @param {Number} reportID
+ * @param {Object} reportAction
+ */
+function deleteReportComment(reportID, reportAction) {
+    // Optimistic Response
+    const reportActionsToMerge = {};
+    const oldMessage = {...reportAction.message};
+    reportActionsToMerge[reportAction.sequenceNumber] = {
+        ...reportAction,
+        message: [
+            {
+                type: 'COMMENT',
+                html: '',
+                text: '',
+            },
+        ],
+    };
+
+    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
+
+    // Try to delete the comment by calling the API
+    API.Report_EditComment({
+        reportID,
+        reportActionID: reportAction.reportActionID,
+        reportComment: '',
+    })
+        .then((response) => {
+            if (response.jsonCode !== 200) {
+                // Reverse Optimistic Response
+                reportActionsToMerge[reportAction.sequenceNumber] = {
+                    ...reportAction,
+                    message: oldMessage,
+                };
+
+                Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
```

Comment:
I think this is only the second time that something is making optimistic changes, other than adding a report comment. I think it's a cool pattern, and probably something that we should try to abstract at some point that would be easier for other actions to use as well. Nothing actionable here, but I wanted to note down the thought.

Reply:
I think that sounds great, there is a certain amount of overhead to do it, so abstraction would be great, specially before we bring the expenses and reports into .cash